lib/rspec/support/method_signature_verifier.rb

```diff
@@ -154,19 +154,57 @@ def classify_parameters
       INFINITY = 1 / 0.0
     end
 
-    # Some versions of JRuby have a nasty bug we have to work around :(.
-    # https://github.com/jruby/jruby/issues/2816
+    # JRuby has only partial support for UnboundMethod#parameters, so we fall back on using #arity
+    # https://github.com/jruby/jruby/issues/2816 and https://github.com/jruby/jruby/issues/2817
     if RSpec::Support::Ruby.jruby? &&
        RubyFeatures.optional_and_splat_args_supported? &&
-       Class.new { attr_writer :foo }.instance_method(:foo=).parameters == []
+       Java::JavaLang::String.instance_method(:char_at).parameters == []
```

Comment:
I'm trying to understand why you changed this conditional, and how this changes things.  Does `Class.new { attr_writer :foo }.instance_method(:foo=).parameters == []` return the same value on all versions of JRuby that `Java::JavaLang::String.instance_method(:char_at).parameters == []` does?

Reply:
As should hopefully be clear by the table in my overall comment, there are no versions where the `attr_writer` bug exists, but the Java proxy method one does not.